core/src/main/scala/org/apache/spark/deploy/rest/MesosRestServer.scala

```diff
@@ -0,0 +1,189 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.spark.deploy.rest
+
+import java.io.File
+import javax.servlet.http.HttpServletResponse
+
+import org.apache.spark.deploy.DriverDescription
+import org.apache.spark.deploy.ClientArguments._
```

Comment:
I'm opposed to just reaching in to the `standalone` space and grabbing default values from there. This is a common theme with this patch that I am not super comfortable with. If we want to reuse code then we should put them in a common place (as you have done by refactoring the `RestServer`) instead of just reaching into the namespace of different packages.